Add the account type to the role names created for SSM Parameter Store access

mcdonnnj opened the pull request:
# <!-- Use the title to describe PR changes in the imperative mood --> #

## 🗣 Description ##

This pull request adds the `-Production` or `-Staging` suffix as appropriate to the roles created by the [cisagov/ssm-read-role-tf-module](https://github.com/cisagov/ssm-read-role-tf-module) module in this configuration. While here I also added the `sts:TagSession` permission to the role assumption document.
<!-- Describe the "what" of your changes in detail. -->
<!-- To avoid scope creep, limit changes to a single goal. -->

## 💭 Motivation and context ##

Right now when using a single account configuration (CyHy Ansible roles) there is a race to see which resources are created first as they share the same name. This fixes that issue and aligns this module's configuration with what is done in [cisagov/ci-iam-user-tf-module](https://github.com/cisagov/ci-iam-user-tf-module) (also used in this configuration) for [Production](https://github.com/cisagov/ci-iam-user-tf-module/blob/1576cd475dd7fe7f0a7550046a377489607e118d/roles.tf#L8) and [Staging](https://github.com/cisagov/ci-iam-user-tf-module/blob/1576cd475dd7fe7f0a7550046a377489607e118d/roles.tf#L19) roles.

We usually add the `sts:TagSession` permission wherever we allow the `sts:AssumeRole` permission so it was likely just missed in this configuration.
<!-- Why is this change required? -->
<!-- What problem does this change solve? How did you solve it? -->
<!-- Mention any related issue(s) here using appropriate keywords such -->
<!-- as "closes" or "resolves" to auto-close them on merge. -->

## 🧪 Testing ##

Automated tests pass. I used this branch in a CyHy Ansible role's Terraform configuration and confirmed that the apply completed successfully and there were no longer errors for already existing resources.
<!-- How did you test your changes? How could someone else test this PR? -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->

<!--
## 📷 Screenshots (if appropriate) ##

Uncomment this section if a screenshot is needed.

-->

## ✅ Pre-approval checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- Draft PRs should have one or more unchecked boxes. -->
<!-- If you're unsure about any of these, don't hesitate to ask. -->
<!-- We're here to help! -->

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

mcdonnnj committed:
Modify the role names for the SSM read roles created

Add `role_name` arguments to the two cisagov/ssm-read-role-tf-module
modules that indicated Production or Staging roles. This prevents
issues with duplicate resource names in a single account scenario and
actually mirrors what is done in the cisagov/ci-iam-user-tf-module
module that is used to create the CI user. The base value for the role
names is the current default for the `role_name` variable as defined
in the cisagov/ssm-read-role-tf-module configuration.

mcdonnnj committed:
Allow session tagging in the role assumption document

It is standard for us to allow `sts:TagSession` permission where we
allow the `sts:AssumeRole` permission.

dav3r reviewed (approved):
Looks good- does this mean that we can now remove the [Single Account Caveats section](https://github.com/cisagov/molecule-iam-user-tf-module#single-account-caveats) from the README?

mcdonnnj committed:
Remove single account usage caveats section from the README

With the change to the role names being created these caveats no longer
exist. Any errors seen should be cause for concern.

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

mcdonnnj commented:
> Looks good- does this mean that we can now remove the [Single Account Caveats section](https://github.com/cisagov/molecule-iam-user-tf-module#single-account-caveats) from the README?

Ah, missed that! Yes this should eliminate that caveats section. Please see [ae1649a](https://github.com/cisagov/molecule-iam-user-tf-module/pull/52/commits/ae1649a2f597dd9a58e5f21db1b782cf7c56638d).
